### PR TITLE
Add missing Override annotations and

### PR DESCRIPTION
**Pull request by croudet**
other code clean-up.

### CODE REVIEW

File: tracing/src/main/java/io/micronaut/tracing/brave/BraveTracerFactory.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ SpanCustomizer spanCustomizer(Tracing tracing) {
     Tracer braveTracer(Tracing tracing) {
         BraveTracer braveTracer = BraveTracer.create(tracing);
         if (!GlobalTracer.isRegistered()) {
-            GlobalTracer.register(braveTracer);
+            GlobalTracer.registerIfAbsent(braveTracer);
```
Comment (jameskleeh) [inline tracing/src/main/java/io/micronaut/tracing/brave/BraveTracerFactory.java:97]: Shouldn't this be replacing the if statement?

File: validation/src/main/java/io/micronaut/validation/validator/DefaultValidator.java [hunk 2]
```diff
@@ -1056,7 +1056,7 @@ public void keyedValue(String nodeName, Object key, Object keyedValue) {
     @SuppressWarnings("unchecked")
     private <T> void validatePojoInternal(@NonNull Class<T> rootClass,
                                           @Nullable T object,
-                                          @NonNull Object[] argumentValues,
+                                          @Nullable Object[] argumentValues,
```
Comment (jameskleeh) [inline validation/src/main/java/io/micronaut/validation/validator/DefaultValidator.java:1059]: What is the reasoning behind this change?

Comment (croudet) [reply validation/src/main/java/io/micronaut/validation/validator/DefaultValidator.java:1059]: https://github.com/micronaut-projects/micronaut-core/blob/8fdb9010e1753af34aece11a9dc1c58539f3cf3e/validation/src/main/java/io/micronaut/validation/validator/DefaultValidator.java#L1122

Calls this method with null for that argument.

File: tracing/src/main/java/io/micronaut/tracing/brave/BraveTracerFactory.java (hunk 1 above)
Comment (croudet) [reply tracing/src/main/java/io/micronaut/tracing/brave/BraveTracerFactory.java]: Done.

**Pull request merged**